clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -692,10 +692,22 @@ private void validateCooperativeAssignment(final Map<String, List<TopicPartition
     }
 
     @Override
-    protected void onJoinPrepare(int generation, String memberId) {
+    protected boolean onJoinPrepare(int generation, String memberId) {
         log.debug("Executing onJoinPrepare with generation {} and memberId {}", generation, memberId);
-        // commit offsets prior to rebalance if auto-commit enabled
-        maybeAutoCommitOffsetsSync(time.timer(rebalanceConfig.rebalanceTimeoutMs));
+        boolean onJoinPrepareAsyncCommitSucceeded = false;
+        // async commit offsets prior to rebalance if auto-commit enabled
+        RequestFuture<Void> future = maybeAutoCommitOffsetsAsync();
+        if (future == null)
+            onJoinPrepareAsyncCommitSucceeded = true;
+        else {
+            if (future.succeeded()) {
+                onJoinPrepareAsyncCommitSucceeded = true;
+            } else if (future.failed() && !future.isRetriable()) {
+                // consistent with async auto-commit failures, we do not propagate the exception
+                log.warn("Asynchronous auto-commit offsets failed: {}", future.exception().getMessage());
+                onJoinPrepareAsyncCommitSucceeded = true;
+            }
```

Comment:
I think we should also log the error in `failed && isRetriable()` case

Reply:
sure